### ISSUE TITLE
i3lock-blur work but flickers.

### ISSUE THREAD

**Issue by in1t3r** [karulont/i3lock-blur#13]
I get a white background and as far as there is no man page with your program I can't really see what the new options mean. If I run i3lock without any other arguments I get just a white background like on the regular i3lock which is not what I want, I dont run composite manager on i3 so that is not a problem. It wouldn't hurt to update a manpage and install it during a make install wouldn't it?
Thanks for your work but this program does not work for me now.

**Comment by karulont** [comment]
You are right: the upstream Makefile does not install the [man page](https://github.com/karulont/i3lock-blur/blob/master/i3lock.1), which is updated by the way.

You should try "i3lock -f"

**Comment by karulont** [comment]
The installation of the man page is a [pull request](https://github.com/i3/i3lock/pull/120) for the upstream i3lock, but has not been merged yet.

If you follow the conversation there, you will see why the man page was not installed.

**Comment by in1t3r** [comment]
Thank you for a fast answer but my main issue was that even after installing all necessary packages for compilation and compiling the i3lock-blur it still does not blur by default but only shows the default white color on the lock screen like the regular i3lock does. 
Are there some parameters I need to pass to i3lock that will make it blur the screen or multiple screens?

**Comment by in1t3r** [comment]
i3lock -f on my system also gives a white background I don't have a compton running nor other composite manager.

**Comment by in1t3r** [comment]
Srry it was my bad the -f argument does work once I rebuild again the i3lock-blur. Thank you for this nice neat version of i3lock. :)

**Comment by in1t3r** [comment]
It still does flicker if I move the mouse or sometime it do a something like a screen rewrite. I do have custom i3bar which is made of conky variables being passed to the same and they are on pretty fast update less then 1 second. Do you know what might be problem its a two monitor setup and its the Intel 5500 graphics.

**Comment by karulont** [comment]
I have Intel HD 4400 graphics, but cannot test with dual monitors at the moment.
I am not sure what causes the flickering, but there is a workaround: "do not redraw the screen, blur only once". It is implemented in the branch "blur-once" and can be enabled with:
"i3lock -fo". Maybe I should merge it in the main branch, if other people find it useful.

**Comment by in1t3r** [comment]
ok thank you that bluronce work similarly to the script that make a single screenshoot and then blur it using imagemagick or something else and set it up as the image background?
So does options -fo just automate that process and basically leave overlay intact? On that way actually it should work even with compton or some other composite manager running? At least I guess so. I'll try the other branch.

**Comment by in1t3r** [comment]
I tried it with compton and actually works pretty fine and I do not have flickers which I have with or without compton or really slow read of pressed keys combinations. Sometime running the master branch with -f option actually make the reading the typed letters really slow it take almost a minute to unlock if it was locked for example more then hour. Not sure where is the problem with it and the bug but it is really problem when you have long complicated passwords like me.
On my opinion blur-once should be merged into the master because it just give one more option and actually allows running of composite window manager on blured i3lock screen. 
There is maybe one more issue but its actually overlay window related issue and that is that if you run compton and run i3lock with -fo blur option it not end up that much blured because compton actually redraws the overlay and you can basically read the terminal text or any other good contrast between text and background. 
I wonder is there are options that can apply on level of blurines of the image. For example do you have that implemented as the i3lock option or you use some default for imagemagick becuase it would be really nice to be able to tweak the fuzzy values and basically pass the arguments to the imagemagick especially for the blur-once branch where it could make a bit more blured image.

**Comment by karulont** [comment]
I made a more readable copy of the [man page](https://gist.github.com/karulont/bf05aa784e7a7c7e04c04d36651ee9a8). You can play with the -r and -s options to adjust the level of blur applied. OpenGL is still used for the blurring, because it is usually a lot faster than imagemagick, but it is done only once.

For the -r value try something like 9 or larger (it depends on screen resolution).
For the -s value start with something like 5 and go higher as needed.
So run with "i3lock -fo -r 9 -s 5".

**Comment by in1t3r** [comment]
thx for the comments issue solved
